DOCUMENTATION.md

```diff
@@ -104,751 +29,740 @@ import uk.gov.service.notify.SendSmsResponse;
+personalisation.put("postcode", "SW14 6BH"); // mandatory address field
+personalisation.put("first_name", "Amala"); // field from template
+personalisation.put("application_date", "2018-01-01"); // field from template
+```
+
+If a template does not have any placeholder fields for personalised information, you must pass in an empty map or `null`.
+
+#### personalisation (optional)
+
+The following parameters in the letter recipient's address are optional:
+
+```java
 personalisation.put("address_line_3", "123 High Street"); // optional address field
 personalisation.put("address_line_4", "Richmond upon Thames"); // optional address field
 personalisation.put("address_line_5", "London"); // optional address field
 personalisation.put("address_line_6", "Middlesex"); // optional address field
-personalisation.put("postcode", "SW14 6BH"); // mandatory address field
-personalisation.put("application_id", "1234"); // field from template
-personalisation.put("application_date", "2017-01-01"); // field from template
-
-SendLetterResponse response = client.sendLetter(templateId, personalisation, "yourReferenceString");
 ```
-</details>
 
-#### Response
+#### reference (optional)
+
+A unique identifier. This reference can identify a single unique notification or a batch of multiple notifications. If you do not have a reference, you must pass in an empty string or `null`.
 
-If the request is successful, the SendLetterResponse is returned from the client. Attributes of the SendLetterResponse are listed below.
+```
+String reference='STRING';
+```
+
+### Response
 
-<details>
-<summary>
-Click here to expand for more information.
-</summary>
+If the request to the client is successful, you will receive a `SendLetterResponse`:
 
 ```java
-	UUID notificationId;
-	Optional<String> reference;
-	UUID templateId;
-	int templateVersion;
-	String templateUri;
-	String body;
-	String subject;
+UUID notificationId;
+Optional<String> reference;
+UUID templateId;
+int templateVersion;
+String templateUri;
+String body;
+String subject;
 ```
 
-Otherwise the client will raise a `NotificationClientException`:
+### Error codes
 
-|`error.status_code`|`error.message`|
-|:---|:---|
-|`429`|`[{`<br>`"error": "RateLimitError",`<br>`"message": "Exceeded rate limit for key type live of 10 requests per 20 seconds"`<br>`}]`|
-|`429`|`[{`<br>`"error": "TooManyRequestsError",`<br>`"message": "Exceeded send limits (50) for today"`<br>`}]`|
-|`400`|`[{`<br>`"error": "BadRequestError",`<br>`"message": "Cannot send letters with a team api key"`<br>`]}`|
-|`400`|`[{`<br>`"error": "BadRequestError",`<br>`"message": "Cannot send letters when service is in trial mode - see https://www.notifications.service.gov.uk/trial-mode"`<br>`}]`|
-|`400`|`[{`<br>`"error": "ValidationError",`<br>`"message": "personalisation address_line_1 is a required property"`<br>`}]`|
+If the request is not successful, the client will raise an `HTTPError`:
```

Comment:
`HTTPError` is a `NotificationClientException`

Reply:
Changed